test/inductor/test_torchinductor.py

```diff
@@ -5244,6 +5244,40 @@ def fn(x):
                 assert same(fn(x)[0], compiled([x])[0], equal_nan=True)
                 assert metrics.generated_cpp_vec_kernel_count == 1
 
+        @unittest.skipIf(
+            not codecache.valid_vec_isa_list(), "Does not support vectorization"
+        )
+        @patch("torch.cuda.is_available", lambda: False)
+        def test_vec_compare_op_cpu_only(self):
+            def fn(x):
+                y1 = torch.eq(x, 1)
+                x = torch.where(y1, x, -x)
```

Comment:
May I know the data type of `1`?

Reply:
The data type of `1` is unclear. I would explicitly define `1` as `float32` by replacing `1` with `1.0`. The same for the other constants.